source/Octopus.Tentacle.Client/Retries/TaskExtensionMethods.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System.Threading.Tasks;
+
+namespace Octopus.Tentacle.Client.Retries
+{
+    static class TaskExtensionMethods
+    {
+        public static void IgnoreUnobservedExceptions(this Task task)
+        {
+            if (task.IsCompleted)
+            {
+                if (task.IsFaulted)
+                {
+                    var observedException = task.Exception;
+                }
+
+                return;
+            }
+
+            task.ContinueWith(
+                t =>
+                {
+                    var observedException = t.Exception;
+                },
+                TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously);
```

Comment:
I wonder if you can use:
```
public static async Task LogAndSwallowAnyException(this Task task, logger) {
    try {
        await task;
    } catch(Exception e) {
        logger.info(e);
    }
}
```
instead.

Reply:
I did a fair bit of reading about it, and this was the approach people took and it seems to work as expected.

I'm not saying awaiting the task wouldn't also work but you may want to ensure it's not a blocking wait (something like `async void LogAndSwallowAnyException`) and I wouldn't log as we aren't really interested in the error.

Reply:
I think just doing the "what everyone recommends" is a good idea, don't change what ain't broke :D